dot/sync/disjoint_block_set_test.go

```diff
@@ -0,0 +1,338 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package sync
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_disjointBlockSet_addBlock(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name             string
+		disjointBlockSet *disjointBlockSet
+		block            *types.Block
+		err              error
+	}{
+		{
+			name:             "add block beyond capacity",
+			disjointBlockSet: &disjointBlockSet{},
+			block: &types.Block{
+				Header: types.Header{
+					Number: 1,
+				},
+			},
+			err: errSetAtLimit,
+		},
+		{
+			name: "add block",
+			disjointBlockSet: &disjointBlockSet{
+				limit:            1,
+				blocks:           make(map[common.Hash]*pendingBlock),
+				timeNow:          time.Now,
+				parentToChildren: make(map[common.Hash]map[common.Hash]struct{}),
+			},
+			block: &types.Block{
+				Header: types.Header{
+					Number: 1,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			err := tt.disjointBlockSet.addBlock(tt.block)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, 1, len(tt.disjointBlockSet.blocks))
+			}
+		})
+	}
+}
+
+func Test_disjointBlockSet_addHeader(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name             string
+		disjointBlockSet *disjointBlockSet
+		header           *types.Header
+		err              error
+	}{
+		{
+			name:             "add header beyond capactiy",
+			disjointBlockSet: &disjointBlockSet{},
+			header: &types.Header{
+				Number: 1,
+			},
+			err: errors.New("cannot add block; set is at capacity"),
+		},
+		{
+			name: "add header",
+			disjointBlockSet: &disjointBlockSet{
+				blocks:           make(map[common.Hash]*pendingBlock),
+				limit:            1,
+				timeNow:          func() time.Time { return time.Unix(1001, 0) },
+				parentToChildren: make(map[common.Hash]map[common.Hash]struct{}),
+			},
+			header: &types.Header{
+				Number: 1,
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			err := tt.disjointBlockSet.addHeader(tt.header)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, 1, len(tt.disjointBlockSet.blocks))
```

Comment:
Can you assert the whole `tt.disjointBlockSet.blocks` 🙏

Reply:
Added check to confirm hash was added to map.